programs/simulator/tests/cases/test_swap_from_dump.rs

```diff
@@ -357,110 +396,108 @@ async fn initialize_accounts(
-                    log::debug!("{log:?}");
-                }
-                None
             } else {
                 None
             }
         }
         Err(e) => {
-            log::warn!("Error simulating : {}", e);
+            log::warn!("Error simulating : {:?}", e);
             None
         }
     }
 }
 
-async fn swap(
-    ctx: &mut ProgramTestContext,
-    owner: &Keypair,
-    instruction: &Instruction,
-) -> anyhow::Result<()> {
-    ctx.get_new_latest_blockhash().await?;
-
-    log::info!("swapping");
-    let result = ctx
-        .sign_send_instructions(&[instruction.clone()], &[&owner])
-        .await;
+async fn swap(ctx: &mut LiteSVM, owner: &Keypair, instruction: &Instruction) -> anyhow::Result<()> {
+    let tx = VersionedTransaction::try_new(
+        VersionedMessage::Legacy(Message::new(&[instruction.clone()], Some(&owner.pubkey()))),
+        &[owner],
+    )
+    .unwrap();
 
+    let result = ctx.send_transaction(tx);
     match result {
-        Ok(()) => Ok(()),
-        Err(e) => Err(anyhow::format_err!("Failed to swap {:?}", e)),
+        Ok(_) => Ok(()),
+        Err(e) => {
+            log::error!("------------- LOGS ------------------");
+            for log in &e.meta.logs {
+                log::error!("{log:?}");
+            }
+            Err(anyhow::format_err!("Failed to swap {:?}", e.err))
+        }
     }
 }
 
 async fn get_balance(
-    ctx: &mut ProgramTestContext,
+    ctx: &mut LiteSVM,
     owner: Pubkey,
     mint: Pubkey,
     is_2022: bool,
 ) -> anyhow::Result<u64> {
     let ata_address = get_associated_token_address(&owner, &mint);
 
-    if is_2022 {
-        let Ok(ata) = ctx.banks_client.get_account(ata_address).await else {
-            return Ok(0);
-        };
-
-        let Some(ata) = ata else {
-            return Ok(0);
-        };
+    let Some(ata) = ctx.get_account(&ata_address) else {
+        return Ok(0);
+    };
 
+    if is_2022 {
         let ata = spl_token_2022::state::Account::unpack(&ata.data);
         if let Ok(ata) = ata {
             return Ok(ata.amount);
         }
     };
 
-    if let Ok(ata) = ctx.get_token_account(ata_address).await {
+    if let Ok(ata) = spl_token::state::Account::unpack(&ata.data) {
         Ok(ata.amount)
     } else {
         Ok(0u64)
     }
 }
 
-async fn set_balance(
-    ctx: &mut ProgramTestContext,
+fn set_balance(
+    ctx: &mut LiteSVM,
     owner: Pubkey,
     mint: Pubkey,
     amount: u64,
     is_2022: bool,
 ) -> anyhow::Result<()> {
-    let ata_address = get_associated_token_address(&owner, &mint);
+    let token_program_id = if is_2022 {
+        spl_token_2022::ID
+    } else {
+        spl_token::ID
+    };
+
+    let ata_address =
+        get_associated_token_address_with_program_id(&owner, &mint, &token_program_id);
+    let mut data = vec![0u8; 165];
 
     if is_2022 {
-        let mut data = vec![0u8; 165];
         let account = spl_token_2022::state::Account {
```

Comment:
add a TODO: to properly setup extensions, this is not sufficient

Reply:
I think it is sufficient for now, because to setup properly the token account with extension we will need much more information like zk stuff right ?